### ISSUE TITLE
Classes/classes.jar not found for jdk 8

### ISSUE THREAD

**Issue by martinschaef** [Sable/soot#686]
Just updated to the nightly build after running an older version of soot for a couple of month. Now I get the exception:
```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: java.lang.Exception: Error: The path '/Library/Java/JavaVirtualMachines/jdk1.8.0_101.jdk/Contents/Home/Classes/classes.jar' does not exist.
	at soot.SourceLocator.getClassSourceType(SourceLocator.java:224)
	at soot.SourceLocator.lookupInClassPath(SourceLocator.java:690)
	at soot.asm.AsmClassProvider.find(AsmClassProvider.java:35)
```
This wasn't an issue before with Java 8. Did you guys change anything? I'm using soot as a library, so I might be missing some setup code that wasn't necessary before.

**Comment by jjsahalf** [comment]
@martinschaef  I am running into the same problems and trying to find a way to walk around.

**Comment by StevenArzt** [comment]
The path looks like you are running Soot on Mac. Does that file exist on Mac? Does the path exist, even if the file does not? What is your full Soot command-line?

**Comment by jjsahalf** [comment]
@StevenArzt Hi Steven. Thank you for your prompt help. I do run it on Mac. I have found a way to solve this. I think there are classes.jar and ui.jar in earlier Java SDK on Mac. Now it just has rt.jar just like Java SDK on any other platforms. I create a 'classes' directory and create soft links classes.jar, ui.jar. Both of them point to rt.jar. So far this way work well for me. Thank you very much again.

**Comment by martinschaef** [comment]
Jep, classes.jar is in java 7 but not in java 8. Fwiw, I did not get this exception with earlier versions of soot.
Btw., since I'm using soot as a library and not through its main, its really complicated to come up with the args that I used. Is there a way to print soot.Options? The toString method is not implemented.

**Comment by loren138** [comment]
Looks like it's not anything new, but this file is probably the issue https://github.com/Sable/soot/blame/8cebfcc9d67addfa0ef8840472df1480654ec62f/src/soot/Scene.java#L627

**Comment by chrisjbaik** [comment]
Along with workaround provided by @jjsahalf , I just did a bit of a hack in my code:

```
System.setProperty("os.name", "Whatever");
Scene.v().loadNecessaryClasses();
System.setProperty("os.name", "Mac OS X");
```

**Comment by martinschaef** [comment]
Thanks @chrisjbaik this does the job 💯 . I didn't even have to add the soft links. Thanks everyone for the useful comments.

**Comment by anddann** [comment]
@mbenz89 fixed the issue, and the release 4.0 works flawless now
